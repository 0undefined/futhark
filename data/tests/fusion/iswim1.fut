-- ==
-- tags { no_opencl }
-- input {
--   [[1,2,3],[4,5,6],[7,8,9]]
-- }
-- output {
--   [[3, 4, 5], [7, 9, 11], [14, 17, 20]]
-- }
-- structure { Map 2 Scan 1 }
fun [[int]] main([[int]] input) =
  let x = scan(fn [int] ([int] a, [int] b) =>
                 map(+, zip(a, b)),
               replicate(3, 0), input) in
  map(fn [int] ([int] r) =>
        map(+ (2), r),
      x)
