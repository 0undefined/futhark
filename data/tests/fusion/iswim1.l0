fun [[int]] main([[int]] input) =
  let x = scan(fn [int] ([int] a, [int] b) =>
                 map(op+, zip(a, b)),
               iota(3), input) in
  map(fn [int] ([int] r) =>
        map(op+ (2), r),
      x)
