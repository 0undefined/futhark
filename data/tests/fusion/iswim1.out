[[3, 5, 7], [7, 10, 13], [14, 18, 22]]
